### PR TITLE
Master pimp my resource cleaned tde

### PR DESCRIPTION
**Pull request by tde-banana-odoo**
Task: https://www.odoo.com/web#id=31320&view_type=form&model=project.task&menu_id=5200

### CODE REVIEW

File: addons/hr/models/hr.py [hunk 1]
```diff
@@ -107,8 +105,12 @@ def _default_image(self):
         image_path = get_module_resource('hr', 'static/src/img', 'default_image.png')
         return tools.image_resize_image_big(open(image_path, 'rb').read().encode('base64'))
 
-    # we need a related field in order to be able to sort the employee by name
-    name_related = fields.Char(related='resource_id.name', string="Resource Name", readonly=True, store=True)
+    # resource
+    name = fields.Char(related='resource_id.name', store=True)
```
Comment (jem-odoo) [inline addons/hr/models/hr.py]: `oldname` can be great here, no ?

File: addons/hr_contract/models/hr_contract.py [hunk 2]
```diff
@@ -61,7 +61,9 @@ class Contract(models.Model):
     date_end = fields.Date('End Date')
     trial_date_start = fields.Date('Trial Start Date')
     trial_date_end = fields.Date('Trial End Date')
-    working_hours = fields.Many2one('resource.calendar', string='Working Schedule')
+    resource_calendar_id = fields.Many2one(
```
Comment (jem-odoo) [inline addons/hr_contract/models/hr_contract.py]: :+1:

File: addons/resource/models/resource_mixin.py [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+        'res.company', 'Company',
+        default=lambda self: self.env['res.company']._company_default_get(),
+        index=True, related='resource_id.company_id', store=True)
+    resource_calendar_id = fields.Many2one(
+        'resource.calendar', 'Working Hours',
+        default=lambda self: self.env['res.company']._company_default_get().resource_calendar_id,
+        index=True, related='resource_id.calendar_id')
+
+    @api.model
+    def create(self, values):
```
Comment (tde-banana-odoo) [inline addons/resource/models/resource_mixin.py]: Actually as name on employee is a stored related it should worr out of the box.

**Comment by tde-banana-odoo** [comment]
Merged at https://github.com/odoo/odoo/commit/64649989bca3bc099e86747b4b81e20f1331ab57 .

**Pull request closed without merging**